File: 卡码/KamaCoder/Python/test11.py
#11. 共同祖先
import sys

def main():
	for line in sys.stdin:
		n=int(line)
		commons=[0 for x in range (21)]
		for i in range(n):
			[a,b]=list(map(int,input().split()))
			commons[a]=b
		height1,height2=0,0
		start1,start2=1,2
		while commons[start1]!=0:
			height1+=1
			start1=commons[start1]
		while commons[start2]!=0:
			height2+=1
			start2=commons[start2]
		if height1>height2:
			print("You are my elder")
		elif height1<height2:
			print("You are my younger")
		else:
			print("You are my brother")

if __name__ == '__main__':
	main()

File: 卡码/KamaCoder/C++/test11.cpp
//11. 共同祖先
#include <iostream>
#include <algorithm>
using namespace std;

int main(){
	int n;
	while(scanf("%d\n",&n)!=EOF){
		int commons[21];
		fill(commons,commons+21,0);
		for(int i=0;i<n;i++){
			int a,b;
			scanf("%d %d\n",&a,&b);
			commons[a]=b;
		}
		int height1=0,height2=0;
		int start1=1,start2=2;
		while(commons[start1]!=0){
			height1++;
			start1=commons[start1];
		}
		while(commons[start2]!=0){
			height2++;
			start2=commons[start2];
		}
		if(height1>height2) printf("You are my elder\n");
		else if(height1<height2) printf("You are my younger\n");
		else printf("You are my brother\n");
	}
	return 0;
}